packages/connection/src/Manager.php

```diff
@@ -43,6 +193,313 @@ public function initialize( $methods ) {
+
+		/**
+		 * Filters the class used for handling XML-RPC requests.
+		 *
+		 * @since 3.1.0
+		 *
+		 * @param string $class The name of the XML-RPC server class.
+		 */
+		$wp_xmlrpc_server_class = apply_filters( 'wp_xmlrpc_server_class', 'wp_xmlrpc_server' );
+		$wp_xmlrpc_server       = new $wp_xmlrpc_server_class();
+
+		// Fire off the request.
+		nocache_headers();
+		$wp_xmlrpc_server->serve_request();
+
+		exit;
+	}
+
+	/**
+	 * Removes all XML-RPC methods that are not `jetpack.*`.
+	 * Only used in our alternate XML-RPC endpoint, where we want to
+	 * ensure that Core and other plugins' methods are not exposed.
+	 *
+	 * @param array $methods a list of registered WordPress XMLRPC methods.
+	 * @return array filtered $methods
+	 */
+	public function remove_non_jetpack_xmlrpc_methods( $methods ) {
+		$jetpack_methods = array();
+
+		foreach ( $methods as $method => $callback ) {
+			if ( 0 === strpos( $method, 'jetpack.' ) ) {
+				$jetpack_methods[ $method ] = $callback;
+			}
+		}
+
+		return $jetpack_methods;
+	}
+
+	/**
+	 * Removes all other authentication methods not to allow other
+	 * methods to validate unauthenticated requests.
+	 */
+	public function require_jetpack_authentication() {
+		// Don't let anyone authenticate.
+		$_COOKIE = array();
+		remove_all_filters( 'authenticate' );
+		remove_all_actions( 'wp_login_failed' );
+
+		if ( $this->is_active() ) {
+			// Allow Jetpack authentication.
+			add_filter( 'authenticate', array( $this, 'authenticate_jetpack' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Authenticates XML-RPC and other requests from the Jetpack Server
+	 *
+	 * @param WP_User|Mixed $user user object if authenticated.
+	 * @param String        $username username.
+	 * @param String        $password password string.
+	 * @return WP_User|Mixed authenticated user or error.
+	 */
+	public function authenticate_jetpack( $user, $username, $password ) {
+		if ( is_a( $user, '\WP_User' ) ) {
+			return $user;
+		}
+
+		$token_details = $this->verify_xml_rpc_signature();
+
+		if ( ! $token_details ) {
+			return $user;
+		}
+
+		if ( 'user' !== $token_details['type'] ) {
+			return $user;
+		}
+
+		if ( ! $token_details['user_id'] ) {
+			return $user;
+		}
+
+		nocache_headers();
+
+		return new \WP_User( $token_details['user_id'] );
+	}
+
+	/**
+	 * Verifies the signature of the current request.
+	 *
+	 * @return false|array
+	 */
+	public function verify_xml_rpc_signature() {
+		if ( is_null( $this->xmlrpc_verification ) ) {
+			$this->xmlrpc_verification = $this->internal_verify_xml_rpc_signature();
+
+			if ( is_wp_error( $this->xmlrpc_verification ) ) {
+				/**
+				 * Action for logging XMLRPC signature verification errors. This data is sensitive.
+				 *
+				 * Error codes:
+				 * - malformed_token
+				 * - malformed_user_id
+				 * - unknown_token
+				 * - could_not_sign
+				 * - invalid_nonce
+				 * - signature_mismatch
+				 *
+				 * @since 7.5.0
```

Comment:
Another case that we should correct to the actual current version we're working towards?

Reply:
Nope, this was indeed introduced in Jetpack 7.5.0.